File: src/main/resources/scripts/tests/interface_test.ts
/// <reference path="imports.d.ts"/>

class interface_test implements IScriptEvents{

    onSave(evt : any){
        out.print("save!");
        var saveData = utils.newMap();
        saveData.put("something", "hi");
        evt.save("interfaceTest", saveData);
    }

    onLoad(evt : any){
        out.print("load!");
        var saveData = evt.load("interfaceTest");
        var s = saveData.get("something");
        out.print(s);
    }

    onReload(evt : any){
    }

}

registerEventHandler(new interface_test());

File: src/main/resources/scripts/tests/interface_test.lua
local events = class('events')

function events.initialize()
end

function events.onSave(EventSave)
    out:print("Lua: Save");
    map = utils:newMap()
    map:put("fromLua", "this is from lua!")
    EventSave:save("lua_interface_test", map)
end

function events.onLoad(EventLoad)
    out:print("Lua: Load");
    map = EventLoad:load("lua_interface_test");
    s = map:get("fromLua");
    out:print(s);
end

registerEventHandler(events:new())

